aten/src/ATen/native/quantized/cpu/qnnpack_fc.cpp

```diff
@@ -0,0 +1,98 @@
+#include "init_qnnpack.h"
+
+#include <ATen/ATen.h>
+#include <ATen/core/Type.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <ATen/quantized/Quantizer.h>
+#include <ATen/Config.h>
+
+namespace at { namespace native {
+namespace {
+
+class QNNPACKFullyConnected final : public c10::OperatorKernel {
+ public:
+  Tensor operator()(at::Tensor input,
+      at::Tensor weight,
+      at::Tensor bias,
+      double output_scale,
+      int64_t output_zero_point) {
+
+    TORCH_CHECK(input.dim() >= 2, "Input tensor rank should be >= 2");
+
+    Tensor input_contig = input.contiguous();
+
+    // C(output) = A(input_contig) x B(weight), where C, A, B are M x N, M x K, N x K
+    // matrices, respectively.
+    int64_t rows_a = input_contig.size(0);
+    int64_t cols_a = 1;
+    for (size_t i = 1; i < input_contig.dim(); ++i) {
```

Comment:
why are you collapsing all dimensions after the first one? I think we should do the opposite - nn.Linear supports input to be arbitrary dimensioned, but only the last dimension participates in matmul: https://pytorch.org/docs/stable/nn.html#linear

Reply:
My bad, I thought it was the other way around. I'll update it to make it consistent with nn.Linear.